cupyx/scipy/sparse/csr.py

```diff
@@ -215,13 +215,38 @@ def eliminate_zeros(self):
         self.indices = compress.indices
         self.indptr = compress.indptr
 
-    def maximum(self, other):
-        # TODO(unno): Implement maximum
+    def _maximum_minimum(self, other, cupy_op, op_name, dense_check):
+        if util.isscalarlike(other):
+            other = cupy.asarray(other, dtype=self.dtype)
+            if dense_check(other):
+                # Note: This is a work-around to make the output dtype the same
+                # as SciPy. It might be SciPy version dependent.
+                dtype = numpy.promote_types(self.dtype, numpy.float64)
+                other = other.astype(dtype)
+                new_array = cupy_op(self.todense(), other)
```

Comment:
We should add a `TODO` comment here about directly comparing the csr matrix with the dense one instead of converting it to dense

Reply:
Actually, the computation steps here are different from those in SciPy. SciPy applies element-wise binary ops (minimum or maximum) to `sparse self` and `sparse other`, whereas here it is `dense self` and `scalar other`. Although this implementation may appear to use more memory, in fact the `sparse other` in SciPy case is essentially a dense matrix, so there is not much difference in memory usage.

In any case, I agree to add a notice that the computation steps here is different from SciPy.